components/movementsensor/dualgps/dualgps.go

```diff
@@ -0,0 +1,231 @@
+package dualgps
+
+import (
+	"context"
+	"math"
+	"sync"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+)
+
+// the default offset between the two gps devices describes a setup where
+// one gps is mounted on the right side of the base and one gps is mounted on the left side of the base
+//   ___________
+//  |	base	|
+//  |			|
+// GPS2			GPS1
+//  |			|
+//  |___________|
+
+const defaultOffsetDegrees = 90.0
+
+var model = resource.DefaultModelFamily.WithModel("dual-gps-rtk")
+
+// Config is used for converting fake movementsensor attributes.
+type Config struct {
+	Gps1   string  `json:"first_gps"`
+	Gps2   string  `json:"second_gps"`
+	Offset float64 `json:"offset_degrees,omitempty"`
+}
+
+func (c *Config) Validate(path string) ([]string, error) {
+	var deps []string
+
+	if c.Gps1 == "" {
+		return nil, resource.NewConfigValidationFieldRequiredError(path, "first_gps")
+	}
+	deps = append(deps, c.Gps1)
+
+	if c.Gps2 == "" {
+		return nil, resource.NewConfigValidationFieldRequiredError(path, "second_gps")
+	}
+	deps = append(deps, c.Gps2)
+
+	return deps, nil
+}
+
+func init() {
+	resource.RegisterComponent(
+		movementsensor.API,
+		model,
+		resource.Registration[movementsensor.MovementSensor, *Config]{Constructor: newDualGPS})
+}
+
+type dualGPS struct {
+	resource.Named
+	logger logging.Logger
+	mu     sync.Mutex
+	offset float64
+
+	gps1 gpsWithLock
+	gps2 gpsWithLock
+}
+
+// newDualGPS makes a new fake movement sensor.
+func newDualGPS(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger logging.Logger,
+) (movementsensor.MovementSensor, error) {
+	dg := dualGPS{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+
+	if err := dg.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return &dg, nil
+}
+
+func (dg *dualGPS) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	dg.mu.Lock()
+	defer dg.mu.Unlock()
+
+	first, err := movementsensor.FromDependencies(deps, newConf.Gps1)
+	if err != nil {
+		return err
+	}
+	dg.gps1.gps = first
+
+	second, err := movementsensor.FromDependencies(deps, newConf.Gps2)
+	if err != nil {
+		return err
+	}
+	dg.gps2.gps = second
+
+	dg.offset = defaultOffsetDegrees
+	if newConf.Offset != 0 {
```

Comment:
Suppose the offset is honestly supposed to be 0 (i.e., one sensor is directly in front of the other): isn't this going to be a problem? A bad workaround is to force the user to swap gps1 with gps2 and make the offset 180, but I think the good way to do this is that the `Config` struct should store the offset as a pointer, so it's either `nil` or a valid number (which might be 0).